src/main/java/bl4ckscor3/mod/scserverutils/mute/PlayerDataManager.java

```diff
@@ -0,0 +1,99 @@
+package bl4ckscor3.mod.scserverutils.mute;
+
+import bl4ckscor3.mod.scserverutils.SCServerUtils;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlayerDataManager {
```

Comment:
Reading/writing the data will be much simpler if you use codecs and NBT. Look at the death logger part of the mod.